### DISCUSSION TITLE
🗺️ How to process the layer before uploading to the BIMS site for harvesting species

### DISCUSSION THREAD

**Discussion by dimasciput** [kartoza/bims#4078] [General]
Jeremy to add more

**Comment by Jeremy-Prior** [comment]
**Generating Processed Layer:**
1. Load layer into QGIS.
2. If multiple features (i.e. feature count > 1); run Vector -> Geometry -> Dissolve. Otherwise, if single feature skip to step 3.
3. Vector -> Data Management -> Reproject layer into a projected CRS (need to be able to work in metres/km, can use 3857).
4. Vector -> Geoprocessing -> Buffer layer: For full countries use ~10 km buffer and for national parks ~5 km (adjust as necessary for feature).
5. Vector -> Geometry -> Simplify features using Douglas-Pecker method and adjusting tolerance as needed (e.g. Full country use tolerance of ~5 km).
6. Vector -> Research -> Create Grid: 
    1. Grid Type: Rectangle
    2. Set Grid CRS to EPSG:3857
    3. Set Horizontal and Vertical spacing to be in km, or metres, per feature. Adjust spacing value per feature it depends on the area of the feature (ensure the value is the same for horizontal and vertical to keep a square grid).
7.  Vector -> Geoprocessing -> Intersection:
    1. Input layer: Simplified layer
    2. Overlay Layer: Grid
8. After you've run the intersection you should aim to have ~16 grid blocks. Rerun step 6 (increasing/decreasing spacing in the grid) and step 7 until you have ~16 grid blocks.
9. Vector -> Geometry -> Polygons to lines.
10. Vector -> Geometry -> Reverse line direction
11. Vector -> Geometry -> Line to polygon.
12. Export/save the layer as geojson.

**Gathering Extent and Centroid:**

1. Right-click on the final/exported layer in the layers panel and view the properties. Copy the extent from the `Information from provider` section.
2. Vector -> Data Management -> Reproject the buffer to EPSG:4326.
3. Vector -> Geometry -> Centroids on the reprojected buffer.
4. Toggle editing on the Centroid, activate the vertex tool, right-click on the point, then copy the X and Y values from the Vertex Editor table (Don't change them, just copy them).
